pkg/tsdb/azuremonitor/applicationinsights-datasource.go

```diff
@@ -250,3 +254,68 @@ func (e *ApplicationInsightsDatasource) getPluginRoute(plugin *plugins.DataSourc
 
 	return pluginRoute, pluginRouteName, nil
 }
+
+// formatApplicationInsightsLegendKey builds the legend key or timeseries name
+// Alias patterns like {{metric}} are replaced with the appropriate data values.
+func formatApplicationInsightsLegendKey(alias string, metricName string, labels data.Labels) string {
+
+	// backwards support for "ToLower" (eww). Could be a problem if there were
+	// two keys that varied only in case, but I don't think that would
+	// happen with azure.
+	lowerLabels := labels.Copy()
```

Comment:
Might it be easier/safer to create an empty data.Labels object, that you copy entries into with lowercased keys? Unsure if it's safe to mutate an array you're iterating over.

Reply:
Mutating a map while iterating is fine in Go. That being said, your suggestion is cleaner - will make the change.